relay.go

```diff
@@ -266,7 +268,7 @@ func (r *Relayer) getDestination(f lazyCallReq, cs relay.CallStats) (*Connection
 	}
 
 	// Get the destination
-	selectedPeer, err := r.hosts.Get(f)
+	selectedPeer, err := r.hosts.Get(f, r.relayConn)
```

Comment:
(only do this if it's not a lot of extra work) Can we add a test to make sure that when the relay calls `Get`, this is not nil.